core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -411,6 +414,13 @@ public SearchResult<WorkflowSummary> search(
                                         return new WorkflowSummary(
                                                 executionDAOFacade.getWorkflowById(
                                                         workflowId, false));
+                                    } catch(ApplicationException ae) {
+                                        if(ApplicationException.Code.NOT_FOUND.equals(ae.getCode())) {
+                                            //if workflow not found  in Redis, remove it from ES
+                                            executionDAOFacade.removeWorkflow(workflowId, false);
```

Comment:
to remove from es, use indexDAO instead of executionDAOFacade. executionDaoFacade may not work as it expects workflow to be present in redis